packages/embedder/ai_models/helpers/fetchWithRetry.ts

```diff
@@ -0,0 +1,61 @@
+import fetch, {RequestInfo, RequestInit, Response} from 'node-fetch'
```

Comment:
I wrote this up real quick to retry in a mechanism similar to how we do API retries with Jira. Note that this supports both retrying at a specific time if `Retry-After` is returned by the server, or falling back to using an exponential backoff.

In either case, the caller must specify a deadline by which the API call must complete

Reply:
Doh! Right! Thank you